lib/colonel_kurtz/block_type.ex

```diff
@@ -57,6 +57,26 @@ defmodule ColonelKurtz.BlockType do
         embeds_one(:content, @content_module)
       end
 
+      @doc """
+      Takes a Block map and converts it to a named BlockType struct according
+      to its :type. Generates a :block_id if necessary, and turns `block.content`
+      into a named BlockType.Content module as well.
+      """
+      @spec from_map(block) :: block_struct
+      def from_map(%{type: type, content: content, blocks: blocks} = attrs) do
```

Comment:
Just a reorg, no changes. `from_map` is the "entrypoint" into this schema so I like seeing it first.